### PR TITLE
docs(aws_sqs source): remove credentials_file from docs

### LINKED ISSUE #11082: `aws_sqs` source: `auth.credentials_file` option does not work
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Vector Version

```
vector 0.19.1 (x86_64-unknown-linux-gnu 3cf70cf 2022-01-25)
```

### Vector Configuration File

Consider the following Vector configuration file:

```yaml
sources:
  my-source:
    type: aws_sqs
    region: YOUR_REGION
    queue_url: YOUR_QUEUE_URL
    auth:
      credentials_file: /path/to/credentials

sinks:
  console:
    type: console
    inputs:
      - my-source
    encoding:
      codec: json
```

### Expected Behavior

According to [the documentation](https://vector.dev/docs/reference/configuration/sources/aws_sqs/#auth.credentials_file) for the `auth.credentials_file` option:

> The path to AWS credentials file. Used for AWS authentication when communicating with AWS services.

If there is a file in `/path/to/credentials` with this content:
```ini
[default]
aws_access_key_id = YOUR_KEY_ID
aws_secret_access_key = YOUR_ACCESS_KEY
```

Then Vector should be able to pick it for authenticating with the AWS services.

### Actual Behavior

The AWS SDK seems to not like to be passed a credentials file (maybe used to but not anymore?) from Vector:
```
2022-01-28T16:45:54.648223Z  WARN vector::aws::aws_sdk::auth: Overriding the credentials file is not supported. `~/.aws/config` and `~/.aws/credentials` will be used instead of "/path/to/credentials"
```

The result is that the credentials file specified in `auth.credentials_file` is completely ignored and then this error appears:
```
2022-01-28T16:50:08.708706Z ERROR vector::sources::aws_sqs::source: SQS receive message error: ConstructionFailure(MissingCredentials)
```

However, the [corresponding environment variable](https://vector.dev/docs/reference/configuration/sources/aws_sqs/#aws_shared_credentials_file) for setting the credentials file **DOES** work correctly, but you have to NOT use `auth.credentials_file` in the Vector configuration itself.
```
AWS_SHARED_CREDENTIALS_FILE=/path/to/credentials vector -c test.yaml
2022-01-28T16:54:02.635898Z  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info,buffers=info"
2022-01-28T16:54:02.636161Z  INFO vector::app: Loading configs. paths=["test.yaml"]
2022-01-28T16:54:03.657047Z  INFO vector::topology::running: Running healthchecks.
2022-01-28T16:54:03.657488Z  INFO vector::topology::running: Starting source. key=my-source
2022-01-28T16:54:03.658073Z  INFO vector::topology::running: Starting sink. key=console
2022-01-28T16:54:03.658676Z  INFO vector: Vector has started. debug="false" version="0.19.1" arch="x86_64" build_id="3cf70cf 2022-01-25"
2022-01-28T16:54:03.659031Z  INFO vector::app: API is disabled, enable by setting `api.enabled` to `true` and use commands like `vector top`.
2022-01-28T16:54:03.666742Z  INFO vector::topology::builder: Healthcheck: Passed.
(...messages from the configured SQS queue appear here...)
```

If the AWS SDK does not allow for a credentials file to be passed anymore, maybe is best to remove this option from Vector altogether to avoid frustration with users. Or maybe there is another way nowadays in the SDK to pass a credentials file?

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #11082 

This removes the `credentials_file` setting from the docs for the aws_sqs source. 

Since all the aws sources share a cue file for the aws settings, I have creating a new one (aws_new_sdk.cue) with the `credentials_file` setting removed that this one uses instead.

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: website/cue/reference/components/aws_new_sdk.cue [hunk 1]
```diff
@@ -0,0 +1,188 @@
+package metadata
+
+components: _aws_new_sdk: {
+	configuration: {
```
Comment (jszwedko) [inline website/cue/reference/components/aws_new_sdk.cue:4]: It seems like only the `auth` configuration differs from `components: _aws_sdk`. Could we maybe inherit the other config and just define `auth` here?

Like:

```
components: _aws_new_sdk: {
  configuration: {
    auth: ...
    endpoint: components._aws_sdk.configuration.endpoint,
}
```

etc.

**Review by jszwedko — COMMENTED** [review]
Thanks! I left one comment below about reducing some of the duplication.

Do we need to update the code too, to disallow this option?

Nit: I think you could just have both of these `_` components defined in the same `aws.cue` rather than a new file.

**Comment by jszwedko** [comment]
> Do we need to update the code too, to disallow this option?

Maybe we could just error on startup if the option is used rather than worry about splitting up all AWS authentication configuration structs.

**Comment by StephenWakely** [comment]
> Maybe we could just error on startup if the option is used rather than worry about splitting up all AWS authentication configuration structs.

Have done this.

```
2022-02-10T10:56:33.924801Z ERROR vector::topology: Configuration error. error=Source "sqs": Overriding the credentials file is not supported.
```

**Review by jszwedko — APPROVED** [review]
Awesome, thanks!

**Pull request merged**